src/views/AmalgamationSelection.vue

```diff
@@ -190,10 +194,21 @@ export default class AmalgamationSelection extends Vue {
 
   /** The amalmagated company result name depending on the type. */
   getRegularAmalgamationText (): string {
-    if (this.isBComp || this.isBcCompany) return 'BC limited company'
-    if (this.isCcc) return 'BC community contribution company'
-    if (this.isUlc) return 'BC unlimited liability company'
-    return 'Unknown'
+    switch (true) {
+      case this.isEntityBcCompany:
+      case this.isEntityBenContinueIn:
+      case this.isEntityBenefitCompany:
+      case this.isEntityContinueIn:
+        return 'BC limited company'
+      case this.isEntityBcCcc:
+      case this.isEntityCccContinueIn:
+        return 'BC community contribution company'
+      case this.isEntityBcUlcCompany:
+      case this.isEntityUlcContinueIn:
+        return 'BC unlimited liability company'
+      default:
+        return 'Unknown'
+    }
```

Comment:
The amalgamation text for the new legal types can't be tested until the allowable actions are updated to allow amalgamation for them.

However, I overrode the allowable actions temporarily to take this screenshot:

![image](https://github.com/bcgov/business-filings-ui/assets/10002889/3c968997-7597-4c7c-9d7c-4e5e60259b7f)